src/main/java/NotebookFactory/ShanghaiNotebookStore.java

```diff
@@ -0,0 +1,18 @@
+package NotebookFactory;
+
+public class ShanghaiNotebookStore extends NotebookStore{
```

Comment:
Where you use it (main or test)? Also I don't see abstract factory. Read https://www.baeldung.com/java-abstract-factory-pattern

Reply:
I deleted all mains since it's not part of pattern